### DISCUSSION TITLE
Single instance for console app?

### DISCUSSION THREAD

**Discussion by sailingbonbini** [orbitalquark/textadept#501] [General]
I would like to use TA in the console, but use an external command to open files in TA. For instance, I'm using a text based file finder, and would like to configure it so that it opens a file in an existing instance of TA running in a different pane. How would I go about this? It looks like the curses version of TA does not have a "single instance" feature ... but would it be possible to somehow connect to the Lua interpreter and run a command through that?

**Comment by orbitalquark** [comment]
The Qt version of Textadept makes use of a third-party Qt library, and the Gtk version of Textadept makes use of a built-in GLib + Dbus feature. I don't know the particulars of how each of them work, but I'd imagine you'd have to do something similar in C for the terminal version.

The problem with doing it in Lua is that you'd need to "listen" for a request and then act upon it. Normally this would cause the application to "block", waiting for such a request. Since Textadept needs to remain responsive to user input, it cannot be blocked in this way. I don't know how to accomplish such a feat :(
